Add remote users when creating a conversation

mdimjasevic opened the pull request:
This adds remote users when creating a conversation. Note: several checks are still missing, as noted in a related funuction `addMembers`:

* A call must be made to the remote backend informing it that this user is now part of that conversation. Use 'updateConversationMemberships'.
-- that call should probably be made after inserting the conversation membership happens in this backend.
* Events should support remote / qualified users, too.

This PR supersedes PR #1528.

pcapriotti reviewed (commented):
Looks good, but it's missing some "fromJSON" golden tests to check that the schema for `NewConv` is backwards compatible, i.e. it can still parse `NewConv` requests that only have unqualified user ids.

I can add that myself later, if that's ok.

mdimjasevic commented:
> Looks good, but it's missing some "fromJSON" golden tests to check that the schema for `NewConv` is backwards compatible, i.e. it can still parse `NewConv` requests that only have unqualified user ids.
> 
> I can add that myself later, if that's ok.

Hm, I thought there is a test that does that. Initially I changed that test, but then someone (Joe?) told me it should still parse even when the new field is not given in JSON, so I reverted the change and the test still passed, i.e., it should be backwards compatible.

At any rate, if you think such a test case is missing, feel free to add it.

mdimjasevic commented on libs/wire-api/test/unit/Test/Wire/API/Golden/FromJSON.hs:
```diff
@@ -29,5 +30,8 @@ tests =
         testFromJSONObjects
-          [(testObject_NewOtrMessage_user_1, "testObject_NewOtrMessage_user_1.json")]
+          [(testObject_NewOtrMessage_user_1, "testObject_NewOtrMessage_user_1.json")],
+      testCase "NewConv" $
+        testFromJSONObjects
+          [(testObject_NewConvUnmanaged_user_1, "testObject_NewConvUnmanaged_user_1.json")]
```
Thank you for this!

pcapriotti commented:
:tada:
